plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -18,43 +14,51 @@
 import static org.junit.Assert.fail;
 
 public class SchemaGenerationTest {
+
+    public static final Logger LOGGER = Logger.getLogger(TokenReloadAction.class.getName());
+    private final String failureMessage  = "The YAML file provided for this schema is invalid";
+
     @Rule
     public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
 
     @Test
     public void validSchemaShouldSucceed() throws Exception {
-        JSONObject schemaObject = generateSchema();
-        JSONObject jsonSchema = new JSONObject(
-            new JSONTokener(schemaObject.toString()));
-        String yamlStringContents = Util.toStringFromYamlFile(this, "validSchemaConfig.yml");
-        JSONObject jsonSubject = new JSONObject(
-            new JSONTokener(Util.convertToJson(yamlStringContents)));
-        Schema schema = SchemaLoader.load(jsonSchema);
+        Schema schema = returnSchema();
         try {
-            schema.validate(jsonSubject);
+            validateSchema(schema, returnJSONSubject("validSchemaConfig.yml"));
         } catch (Exception e) {
+            LOGGER.warning(failureMessage);
             fail(e.getMessage());
         }
     }
 
     @Test
     public void invalidSchemaShouldNotSucceed() throws Exception {
+        Schema schema = returnSchema();
+        try {
+            validateSchema(schema, returnJSONSubject("invalidSchemaConfig.yml"));
+        } catch (Exception e) {
+            LOGGER.warning(failureMessage);
+            fail(e.getMessage());
+        }
+    }
+
+    private Schema returnSchema() {
         JSONObject schemaObject = generateSchema();
-        Gson gson = new GsonBuilder().setPrettyPrinting().create();
-        JsonParser jsonParser = new JsonParser();
-        JsonElement jsonElement = jsonParser.parse(schemaObject.toString());
-        String prettyJsonString = gson.toJson(jsonElement);
         JSONObject jsonSchema = new JSONObject(
-            new JSONTokener(prettyJsonString));
-        String yamlStringContents = Util.toStringFromYamlFile(this, "invalidSchemaConfig.yml");
+            new JSONTokener(schemaObject.toString()));
+        Schema schema = SchemaLoader.load(jsonSchema);
+        return schema;
+    }
+
+    private JSONObject returnJSONSubject(String yamlFile) throws Exception {
+        String yamlStringContents = Util.toStringFromYamlFile(this, yamlFile);
         JSONObject jsonSubject = new JSONObject(
             new JSONTokener(Util.convertToJson(yamlStringContents)));
-        Schema schema = SchemaLoader.load(jsonSchema);
-        try {
-            schema.validate(jsonSubject);
-            fail();
-        } catch (ValidationException ve) {
-            ve.printStackTrace();
-        }
+        return jsonSubject;
+    }
+
+    private void validateSchema(Schema schema, JSONObject jsonSubject) throws Exception {
```

Comment:
this should probably return a boolean, then you can  assert true on it this method

Reply:
This should probably belong in the Utils class as well for re-usability and so that other plugins can use it.

Not sure if Schema should be passed as a parameter to it as well